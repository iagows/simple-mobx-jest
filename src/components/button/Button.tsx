import React, { FunctionComponent } from "react";

interface IButton {
  onClick: () => void;
  value: string;
}

const Button: FunctionComponent<IButton> = ({
  onClick,
  value,
}: React.PropsWithChildren<IButton>) => {
  return (
    <button className="bt" onClick={() => onClick()}>
      {value}
    </button>
  );
};

export default Button;
