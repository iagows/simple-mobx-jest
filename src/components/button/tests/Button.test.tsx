import React from "react";

import { shallow } from "enzyme";

import Button from "../Button";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const _onClickMock = jest.fn();

const text: string = "botão";

describe("<Button>", () => {
  const props = {
    onClick: _onClickMock,
    value: text,
  };

  const buttonWrapper = shallow(<Button {...props} />);

  describe("When mounted", () => {
    it("should have snapshot", () => {
      expect(buttonWrapper).toMatchSnapshot();
    });
  });

  it("testando o conteúdo", () => {
    const content = buttonWrapper.find(".bt");

    expect(content.props().children).toBe(text);
  });

  it("testando a função", () => {
    buttonWrapper.simulate("click");

    expect(_onClickMock).toBeCalledTimes(1);
  });
});
