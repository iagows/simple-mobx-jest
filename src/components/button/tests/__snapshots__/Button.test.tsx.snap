// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> When mounted should have snapshot 1`] = `ShallowWrapper {}`;
